src/lowering/desugar.ml

```diff
@@ -703,7 +705,8 @@ and obj obj_typ efs bases =
   let ds, fs = map (exp_field obj_typ) efs |> split in
   let ds', fs' = concat_map gap (T.as_obj obj_typ |> snd) |> split in
   let obj_e = newObjE T.Object (append fs fs') obj_typ in
-  I.BlockE(append base_decs (append ds ds'), obj_e)
+  let decs = append base_decs (append (filter_map Fun.id ds) ds') in
+  (blockE decs obj_e).it
```

Comment:
```suggestion
  let decs = append base_decs (append (flatten ds) ds') in
  (blockE decs obj_e).it
```
(and rename this ds to dss since its now a nested list?)